crates/wasi-common/src/snapshots/preview_1.rs

```diff
@@ -1115,22 +1112,18 @@ impl wasi_snapshot_preview1::WasiSnapshotPreview1 for WasiCtx {
             // `&mut [u8]`'s exclusive ownership which is not guaranteed due to
             // potential access from other threads).
             let mut tmp = vec![0; buf.len().min(MAX_SHARED_BUFFER_SIZE)];
-            self.random.try_fill_bytes(&mut tmp)?;
+            self.random.lock().unwrap().try_fill_bytes(&mut tmp)?;
```

Comment:
Could you file an issue and list a TODO here for "we shouldn't serialize calls to `random_get`"?

Reply:
It felt strange to create an issue for yet-to-be-merged code; I will do this as a follow-up.